xapian-bindings/python3/utilletor.i

```diff
@@ -0,0 +1,165 @@
+    if (!SWIG_IsOK(res)) {
+	if (res == INT_MIN) SWIG_fail;
+	%argument_fail(res, "$type", $symname, $argnum);
+    }
+    if (!ptr) {
+	%argument_nullref("$type", $symname, $argnum);
+    }
+    $1 = ptr;
+}
+%typemap(in, fragment="XapianSWIG_anystring_as_ptr") std::string {
+    std::string *ptr = (std::string *)0;
+    int res = XapianSWIG_anystring_as_ptr($input, &ptr);
+    if (!SWIG_IsOK(res) || !ptr) {
+	if (res == INT_MIN) SWIG_fail;
+	%argument_fail((ptr ? res : SWIG_TypeError), "$type", $symname, $argnum);
+    }
+    $1 = *ptr;
+    if (SWIG_IsNewObj(res)) delete ptr;
+}
+%typemap(freearg, noblock=1, match="in") const std::string & {
+    if (SWIG_IsNewObj(res$argnum)) %delete($1);
+}
+%typemap(typecheck, noblock=1, precedence=900) const std::string & {
+    if (PyUnicode_Check($input)) {
+	$1 = 1;
+    } else if (PyBytes_Check($input)) {
+	$1 = 1;
+    } else {
+	int res = SWIG_AsPtr_std_string($input, (std::string**)(0));
+	$1 = SWIG_CheckState(res);
+    }
+}
+
+%typemap(in, fragment="XapianSWIG_anystring_as_ptr") const std::string *(int res = SWIG_OLDOBJ) {
+    std::string *ptr = (std::string *)0;
+    if ($input != Py_None) {
+	res = XapianSWIG_anystring_as_ptr($input, &ptr);
+	if (!SWIG_IsOK(res)) {
+	    if (res == INT_MIN) SWIG_fail;
+	    %argument_fail(res, "$type", $symname, $argnum);
+	}
+    }
+    $1 = ptr;
+}
+%typemap(freearg, noblock=1, match="in") const std::string * {
+    if (SWIG_IsNewObj(res$argnum)) %delete($1);
+}
+%typemap(typecheck, noblock=1, precedence=900) const std::string * {
+    if ($input == Py_None) {
+	$1 = 1;
+    } else if (PyUnicode_Check($input)) {
+	$1 = 1;
+    } else if (PyBytes_Check($input)) {
+	$1 = 1;
+    } else {
+	int res = SWIG_AsPtr_std_string($input, (std::string**)(0));
+	$1 = SWIG_CheckState(res);
+    }
+}
+
+/* This typemap is only currently needed for returning a value from the
+ * get_description() method of a Stopper subclass to a C++ caller, but might be
+ * more generally useful in future.
+ */
+%typemap(directorout, noblock=1, fragment="XapianSWIG_anystring_as_ptr") std::string {
+    std::string *swig_optr = 0;
+    int swig_ores;
+    {
+	PyObject * tmp = $input;
+	Py_INCREF(tmp);
+	swig_ores = XapianSWIG_anystring_as_ptr(tmp, &swig_optr);
+	Py_DECREF(tmp);
+    }
+    if (!SWIG_IsOK(swig_ores) || !swig_optr) {
+	%dirout_fail((swig_optr ? swig_ores : SWIG_TypeError), "$type");
+    }
+    $result = *swig_optr;
+    if (SWIG_IsNewObj(swig_ores)) %delete(swig_optr);
+}
+
+%typemap(directorin) (size_t num_tags, const std::string tags[]) {
+    PyObject * result = PyList_New(num_tags);
+    if (result == 0) {
+	return NULL;
+    }
+
+    for (size_t i = 0; i != num_tags; ++i) {
+	PyObject * str = PyBytes_FromStringAndSize(tags[i].data(), tags[i].size());
+	if (str == 0) {
+	    Py_DECREF(result);
+	    return NULL;
+	}
+
+	PyList_SET_ITEM(result, i, str);
+    }
+    $input = result;
+}
+
+/* vim:set syntax=cpp:set noexpandtab: */
```

Comment:
This looks like it's just a copy of `util.i`, complete with several typemaps letor doesn't currently need.

Perhaps it would be better to simply include `util.i` directly rather than forking our own code needlessly and creating extra maintenance work for our future selves?

Reply:
fixed in a fixup commit.